src/iexamine.cpp

```diff
@@ -1836,7 +1837,7 @@ void iexamine::kiln_full(player &, const tripoint &examp)
     }
 
     item result( "charcoal", calendar::turn.get_turn() );
-    result.charges = total_volume * char_type->ammo->def_charges / char_type->volume;
+    result.charges = total_volume / char_type->volume;
```

Comment:
I suspect this is a pre-existing bug, it should have used `stack_size` in the first place. Or should have explained why it uses `def_charges`.

Currently charcoal has `def_charges` 50 and `stack_size` 10. A stack of 20 unfinished charcoal (volume 20), created from 5 logs (volume 200) will transform into 1000 charcoal (volume 100). The last transformation (volume 20 -\> volume 100) seems wrong.

This should probably just call `result.charges_per_volume( total_volume )`, which does the same as the new code here. The new logic here preserves the volume of the item.